encoding/bcd.go

```diff
@@ -45,3 +48,31 @@ func (e *bcdEncoder) Decode(src []byte, length int) ([]byte, int, error) {
 	// e.g. 0643 => 643
 	return dst[decodedLen-length:], read, nil
 }
+
+func (e bcdEncoder) DecodeFrom(r io.Reader, length int) (data []byte, read int, err error) {
```

Comment:
I don’t like this method. I’m thinking about keeping Decode only and find other way to read data for decoding.

Reply:
Similar to this I was wondering why an `Encoder` interface contained "Decode" methods.

Reply:
Adam, I named it so. I don't see how we can separately use Encoder and Decoder. They always go in pairs and in the `Spec` you specify one thing that will be both used for encoding and decoding.

I just named it `Coder` to avoid confusion.